examples/simple-game-server/helper.go

```diff
@@ -0,0 +1,331 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	coresdk "agones.dev/agones/pkg/sdk"
+	sdk "agones.dev/agones/sdks/go"
+)
+
+// ready attempts to mark this gameserver as ready
+func ready(s *sdk.SDK) {
+	err := s.Ready()
+	if err != nil {
+		log.Fatalf("Could not send ready message")
+	}
+}
```

Comment:
So I am noting that we have a pattern that looks like the above quite a bit - execute a function on the SDK, if it fails, `log.Fatalf(...)`

Not 100% sure how useful this would be - as it may only apply in a few places, but something like:

```go
func runSDK(f func() err, failMsg string) {
   if err := f(); err != nil {
      log.Fatalf(failMsg + ": %v", err)
   }
}
```

Then this function could pretty much go away and you could do inline in `handleReady`

```go
func handleReady(s *sdk.SDK, parts []string, _ ...context.CancelFunc) (response string, addACK bool, responseError error) {
   runSDK(func() err { return s.Ready() }, "Could not send ready message")
}
```

Since it's a closure, it could also handle anything taking in a series of arguments.

Where it goes fall over is if a SDK function returns several return values -- so it may not be worth it. But figured I'd show you my general idea, and see if it made sense -- save writing two functions for each command.

Reply:
> So I am noting that we have a pattern that looks like the above quite a bit - execute a function on the SDK, if it fails, `log.Fatalf(...)`
> 
> Not 100% sure how useful this would be - as it may only apply in a few places, but something like:
> 
> ```go
> func runSDK(f func() err, failMsg string) {
>    if err := f(); err != nil {
>       log.Fatalf(failMsg + ": %v", err)
>    }
> }
> ```
> 
> Then this function could pretty much go away and you could do inline in `handleReady`
> 
> ```go
> func handleReady(s *sdk.SDK, parts []string, _ ...context.CancelFunc) (response string, addACK bool, responseError error) {
>    runSDK(func() err { return s.Ready() }, "Could not send ready message")
> }
> ```
> 
> Since it's a closure, it could also handle anything taking in a series of arguments.
> 
> Where it goes fall over is if a SDK function returns several return values -- so it may not be worth it. But figured I'd show you my general idea, and see if it made sense -- save writing two functions for each command.

Ahan! I got it. I have added all the sdk functions inside the handlers.